pkg/kubelet/cm/cpumanager/cpu_manager.go

```diff
@@ -225,8 +225,8 @@ func (m *manager) reconcileState() (success []reconciledContainer, failure []rec
 	for _, pod := range m.activePods() {
 		allContainers := pod.Spec.InitContainers
 		allContainers = append(allContainers, pod.Spec.Containers...)
+		status, ok := m.podStatusProvider.GetPodStatus(pod.UID)
 		for _, container := range allContainers {
-			status, ok := m.podStatusProvider.GetPodStatus(pod.UID)
 			if !ok {
```

Comment:
If we do make this change (and I'm not yet positive its safe), I believe we should also move this `if !ok` check outside of the loop.

Reply:
The variable reference is to the ok outside the container loop.
The intention of the check here is to include container name in the failure.

So the check should be kept here.

Reply:
I agree with @mattjmcnaughton on this one, in the interest of local readability, if `ok` is false you could immediately append all of the containers to the failure list `continue` the outer pod iteration loop. As is, only the first container is logged; it makes more sense to just log the pod name.

Reply:
Please confirm that the test needs to be modified when only pod name is returned:
```
			if !foundFailedContainer {
				t.Errorf("Expected reconciliation failure for container: %s", testCase.expectFailedContainerName)
```

Reply:
Meaning, expectFailedContainerName parameter can be dropped from cpu_manager_test.go

Reply:
@ConnorDoyle 
If you think the potential changes to test are large, we can use the current formation where compatibility with current code is kept.

Thanks

Reply:
Let's keep the reconciledContainer struct as-is; there are other scenarios that need to test for individual container reconcile failures.

For this change, if the pod status can't be queried we loop over each container in the pod and add it to the list of failures. I don't believe the tests would need to change beyond expecting all of the container to be in the failed list in this scenario.

Let me know if that isn't clear.

Reply:
I updated the PR. Let me know what you think, @ConnorDoyle 

In order not to have two if blocks with same condition, I kept the if block inside the container loop.

Reply:
This is no worse than what we had before, so if you don't feel like making further changes it's OK. Just for clarity, I had something like this in mind:

```go
	for _, pod := range m.activePods() {
		allContainers := pod.Spec.InitContainers
		allContainers = append(allContainers, pod.Spec.Containers...)
		if status, ok := m.podStatusProvider.GetPodStatus(pod.UID); !ok {
			klog.Warningf("[cpumanager] reconcileState: skipping pod; status not found (pod: %s)", pod.Name)
			for _, container := range allContainers {
				failure = append(failure, reconciledContainer{pod.Name, container.Name, ""})
			}
			continue  // to next pod
		}

		for _, container := range allContainers {
			// ...
		}
```

The benefit I see it to include all containers from the pod we failed to get status in the failure list, instead of just the first container.

Reply:
The current code is consistent with previous behavior.

So please accept the PR if it is Okay with you.

Thanks

Reply:
Usually for such a small efficiency gain the readability hit would not be worth it. It's idiomatic to check error conditions immediately in go. But, since it's only one line removed it is OK.